### ISSUE TITLE
TLS connection issues

### ISSUE THREAD

**Issue by defanor** [jorgenschaefer/circe#280]
As was mentioned in IRC, when a remote server isn't available, in some
cases Circe fails to recognize the failed gnutls attempt to connect,
and stucks on "Connecting". To reproduce it, drop the packets with
`iptables`, like `iptables -I OUTPUT -d tepper.freenode.net -j DROP`,
and try to connect or reconnect.

**Comment by jorgenschaefer** [comment]
Thank you for the report!

**Comment by wasamasa** [comment]
Can you still reproduce this after the fix in https://github.com/jorgenschaefer/circe/issues/377 ?

**Comment by wasamasa** [comment]
Uh-oh, `gnutls-boot-parameters` seems to be an Emacs 26.1 thing. I thought it was in `gnutls.el` since the beginning, apparently not.

**Comment by wasamasa** [comment]
I've pushed a fix, could you please give it a try?

**Comment by defanor** [comment]
Indeed, now it works here as it used to: no elisp errors, but no
reconnect attempts (after 25 minutes) either.

**Comment by wasamasa** [comment]
If you can test with 26.1 or newer, I'd be curious whether that resolves it. At this point I doubt it's TLS-related at all and https://github.com/jorgenschaefer/circe/pull/372 could be related.

**Comment by defanor** [comment]
Tried it on 27.1, after 2 minutes the "[conn.failed from nil]" message
appeared. Also tried to connect with lagmon on, and to start dropping
packets mid-connection: after a minute Circe attempted to reconnect,
after 2 more minutes there was the "[conn.failed from nil]" message, and
no further reconnect attempts (waited for 10+ minutes).

**Comment by wasamasa** [comment]
Now for the most interesting part: Does this happen when not using TLS, too? If yes, then I'd close this in favor of some other issue/PR.

**Comment by defanor** [comment]
> Now for the most interesting part: Does this happen when not using
> TLS, too? If yes, then I'd close this in favor of some other issue/PR.

Yes, appears to be the same without TLS: "[conn.failed from nil]", and
no reconnect.

**Comment by wasamasa** [comment]
Hm, there doesn't appear to really be a non-TLS issue for this kind of hang, https://github.com/jorgenschaefer/circe/issues/226 is the closest. Maybe just rename this one?
